core/src/main/java/org/apache/accumulo/core/security/crypto/NonCachingSecretKeyEncryptionStrategy.java

```diff
@@ -81,7 +81,10 @@ private void doKeyEncryptionOperation(int encryptionMode, CryptoModuleParameters
       Cipher cipher = DefaultCryptoModuleUtils.getCipher(params.getAllOptions().get(Property.CRYPTO_DEFAULT_KEY_STRATEGY_CIPHER_SUITE.getKey()));
 
       try {
-        cipher.init(encryptionMode, new SecretKeySpec(keyEncryptionKey, params.getAlgorithmName()));
+    	if (keyEncryptionKey.length > 0)
```

Comment:
This makes me wonder what the API contract is on `read()`. Will the return value from this always be equal to the length of the byte array? If it's not, should we explicitly throw an `InvalidKeyException`? Should we use some `readFully()` method to populate the read buffer instead of `read()`?